R/Design-methods.R

```diff
@@ -4710,3 +4710,311 @@ setMethod("simulate",
 
 ## --------------------------------------------------------------------------
 # nolint end
+
+# simulate ----
+
+## DesignGrouped ----
+
+#' Simulate Method for the [`DesignGrouped`] Class
+#'
+#' @description `r lifecycle::badge("experimental")`
+#'
+#' A simulate method for [`DesignGrouped`] designs.
+#'
+#' @param object (`DesignGrouped`)\cr the design we want to simulate trials from.
+#' @param nsim (`number`)\cr how many trials should be simulated.
+#' @param seed (`RNGstate`)\cr generated with [setSeed()].
+#' @param truth (`function`)\cr a function which takes as input a dose (vector) and
+#'   returns the true probability (vector) for toxicity for the mono arm.
+#'   Additional arguments can be supplied in `args`.
+#' @param combo_truth (`function`)\cr same as `truth` but for the combo arm.
+#' @param args (`data.frame` or `NULL`)\cr optional `data.frame` with arguments that work
+#'   for the `truth` and `combo_truth` functions. The column names correspond to the argument names, the rows to
+#'   the values of the arguments. The rows are appropriately recycled in the `nsim`
+#'   simulations. In order to produce outcomes from the posterior predictive
+#'   distribution, e.g, pass an `object`that contains the data observed so
+#'   far, `truth` contains the `prob` function from the model in
+#'   `object`, and `args` contains posterior samples from the model.
+#' @param firstSeparate (`flag`)\cr whether to enroll the first patient separately
```

Comment:
we will do that later when refactoring the whole `simulate` methods, right now keeping this for consistency (because this PR is not intended for general cleanup, just adding new features)